File: resizer/src/main/java/com/trailmagic/resizer/ImageResizeService.java
package com.trailmagic.resizer;

import java.io.File;
import java.io.InputStream;
import java.util.List;

public interface ImageResizeService {
    public List<ImageFileInfo> scheduleResize(InputStream srcInputStream) throws ResizeFailedException;
    public List<ImageFileInfo> scheduleResize(File srcFile) throws ResizeFailedException;
    public ImageFileInfo identify(File srcFile);
    public File writeFile(InputStream srcInputStream) throws ResizeFailedException;
}

File: resizer/src/main/scala/com/trailmagic/resizer/ImageResizeService.scala
package com.trailmagic.resizer

import java.io.File
import java.io.InputStream
import java.util.List

trait ImageResizeService {
  def scheduleResize(srcInputStream: InputStream): List[ImageFileInfo]

  def scheduleResize(srcFile: File): List[ImageFileInfo]

  def identify(srcFile: File): ImageFileInfo

  def writeFile(srcInputStream: InputStream): File
}